code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -237,17 +239,50 @@
 			LAZYREMOVE(created_items, ref)
 
 	if(LAZYLEN(created_items) >= limit)
-		loc.balloon_alert(user, "ritual failed, at limit!")
-		return FALSE
+		for(var/obj/item/melee/sickly_blade/is_blade_ritual as anything in result_atoms)
+			valid_blades = blades_limit_check(user)
+			break
+		if(valid_blades)
+			return TRUE
+		else
+			loc.balloon_alert(user, "ritual failed, at limit!")
+			return FALSE
 
 	return TRUE
 
 /datum/heretic_knowledge/limited_amount/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
 	for(var/result in result_atoms)
 		var/atom/created_thing = new result(loc)
 		LAZYADD(created_items, WEAKREF(created_thing))
+		if(istype(created_thing, /obj/item/melee/sickly_blade))
+			add_to_list_sickly_blade(user, created_thing)
 	return TRUE
 
+/datum/heretic_knowledge/limited_amount/proc/add_to_list_sickly_blade(mob/living/heretic, obj/item/melee/sickly_blade/created_blade)
+	var/obj/item/melee/sickly_blade/blade_check = created_blade
+	var/datum/antagonist/heretic/our_heretic = IS_HERETIC(heretic)
+	if(!isnull(our_heretic))
+		blade_check.owner = our_heretic
+		LAZYADD(our_heretic.blades_list, blade_check)
+
+/datum/heretic_knowledge/limited_amount/proc/blades_limit_check(mob/living/heretic)
+	var/datum/antagonist/heretic/our_heretic = IS_HERETIC(heretic)
+	var/success_check = FALSE
+	for(var/obj/item/melee/sickly_blade/blades_in_list in our_heretic.blades_list)
+		if(get_turf(heretic) == get_turf(blades_in_list))
+			continue
+		success_check = TRUE
+		LAZYREMOVE(our_heretic.blades_list, blades_in_list)
+		for(var/mob/living/living_target in range(0, get_turf(blades_in_list)))
```

Comment:
```suggestion
		var/mob/living/living_target = recursive_loc_check(src, mob/living)
		if(living_target)
```
Items can only be held by one mob at a time so I don't think we should check for every mob on its location...